src/alignment/Graph.h

```diff
@@ -1,29 +1,67 @@
 #pragma once
 
-#include <fstream>
+#include<fstream>
 #include<string>
 #include<vector>
 #include<array>
 #include<fstream>
+#include "Node.h"
+#include<utility>
+
+using std::vector;
+using std::string;
 
 class Graph {
-	private:
-		int k;
-		
-		std::vector<std::array<int,3>> listOfEdges;					// vector of edges (as array with size 3)
-		std::vector<std::string> stringListSequence;				// vector of strings for the sequences
-		std::vector<int> numberOfKmer;								// vector amount of k-mer
-		// std::vector<std::vector<int>> adjacentEdges;
-		void calcEdgeList(); 								        // Method: calculate list of edges
-		
-	public:
-	    Graph(int k);											    // constructor
+    private:
+        
+        int k;                                     ///< length of any node (kmer)
+
+        /** vector of adjacency edges for each 
+         *  node in each sequence
+         */
+        vector<vector<vector<Node> >> listOfEdges;

```

Comment:
The variable name got me confused. It is rather a list of adjacent nodes than a list of edges. I suggest to rename it `adjacentNodes`, instead. "list" does not need to be part of the name.
The bigger issue is, however, that we should have a data structure "Edge". When drawn edges have properties, e.g. they are highlighted when hovered over. We select edges in the game and edges can be admissible or not. If you want to attach properties to your implicitly defined "edges", you would have to maintain parallel vectors.
It is more natural and even simpler to let a graph have a set of edges and nodes as in standard graph notation. More on this Tuesday.